include/utils.hpp

```diff
@@ -37,9 +37,113 @@ namespace utils {
      * The cheap way to fix it is to limit rank to 99
     */
 
-    uint64_t r = (current * 100) / total; 
-    if (r > 99) return 99;
-    return r;
+    float rank_coefs[99] = { 0.00,
```

Comment:
Another issue - this spline ranks array should contain 100 values - from 0 - 99

The possible values for the _index_ are between 0 and 99. 

So that's 100 values, not 99.

Reply:
I added some test code that tests this and prints out things

Reply:
This is fixed.